### DISCUSSION TITLE
Load extensions in Firefox for Safe Exam Browser

### DISCUSSION THREAD

**Discussion by Svangeel89** [SafeExamBrowser/seb-win-refactoring#900] [Q&A]
Hello everyone,

I can't find any information about it anywhere, so I'm asking everyone here.

Situation: I have created some extensions that perform/manipulate a number of things such as:

- Adjust HTTP headers so that everything can be loaded within frames.
- View user screens with tabs.captureVisibleTab()
- Adblocker and DOM manipulation so that every external link can be loaded in the same window
- ....

For our ChromeOS enterprise devices I can easily load these extensions within Google Workspace (kiosk mode), but for our Windows devices we use Safe Exam Browser.

Now the question is how and whether it is technically possible to load my extensions in the (Firefox) browser within Safe Exam Browser.

Does anyone have experience or concrete advice on how to get started with this?

Thank you in advance for your time!

**Comment by dbuechel** [comment]
SEB 3.x uses Chromium, it was the legacy version 2.x which was using Firefox as integrated browser engine. That said, you could certainly use SEB without the integrated browser application and add Firefox as permitted application to then run your web application there (see https://safeexambrowser.org/windows/win_usermanual_en.html#ApplicationsPane).

**Reply by Svangeel89** [reply]
Thanks for your comment! Good to know, I assumed it was still Firefox. However, I would like to use the built-in browser so that one can only use our own web application and cannot navigate further. The fact that this is Chromium makes it simple for me because the current extensions are written specifically for Chrome. But the question remains whether extensions can be loaded into Safe Exam's built-in browser.

**Reply by dbuechel** [reply]
No, unfortunately we do currently not support extensions and it is unlikely that we ever will.
